### PR TITLE
add watchdog/job-exporter

### PR DESCRIPTION
**Pull request by xudifsd**
Working in progress.

TODOs:

- [x] Current hard code some IP and config, should change it later.
- [x] Currently using xudifsd/job-exporter etc. Should change it to openpai/job-exporter:dlts.

### CODE REVIEW

**Comment by xudifsd** [comment]
Still hard coded prometheus IP:PORT in grafana config.

File: src/ClusterBootstrap/services/monitor/grafana-config.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
+{% set prometheus_url = "192.168.255.1:9091" %}
```
Comment (hongzhili) [inline src/ClusterBootstrap/services/monitor/grafana-config.yaml]: How about "{{ cnf["api-server-ip"] }}:9091" ?

Comment (xudifsd) [reply src/ClusterBootstrap/services/monitor/grafana-config.yaml]: This assumes api-server is always co-allocated with prometheus, we can assume that?

**Comment by xudifsd** [comment]
@hongzhili I just saw there are many dashboards available in old Grafana, some of them are still in use, for example `GPU usage` from `Cluster Status`, do I need to import them? If we import them, we will need to keep old influex & collected.

File: src/ClusterBootstrap/services/monitor/grafana-config.yaml (hunk 1 above)
Comment (xudifsd) [reply src/ClusterBootstrap/services/monitor/grafana-config.yaml]: Have tried this, the value will be `10.3.0.1` in dltsp40 cluster, the grafana pod can not connect to this address, not sure what this IP is.

**Comment by xudifsd** [comment]
There will be two dashboard, one:

image

another:
image

There are several problems:
* Current available gpu is get by watchdog requesting k8s's node api, I submitted a job requiring 1 gpu, and job is running, but available GPU from k8s's node api doesn't change, I'm not sure why.
* Original `Cluster Status` have two more columns: `Reserved GPU` and `Total Jobs`, not sure how to get these two numbers.

**Comment by xudifsd** [comment]
From k8s issues https://github.com/kubernetes/kubernetes/issues/43045 and [doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-allocatable.md), the allocatable is defined as:

```
[Allocatable] = [Node Capacity] - [Kube-Reserved] - [System-Reserved] - [Hard-Eviction-Threshold]
```

It's not available resources. I will find another way to do this.

File: src/ClusterBootstrap/services/monitor/pre-render.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+dir=`dirname $0`
+
+file_name=${dir}/grafana-config.yaml
+
+# create configmap
+for i in `find ${dir}/grafana-config/ -type f -regex ".*json" ` ; do
+    echo --from-file=$i
+done | xargs ${dir}/../../deploy/bin/kubectl --namespace=kube-system create configmap grafana-configuration --dry-run -o yaml >> $file_name
```
Comment (hongzhili) [inline src/ClusterBootstrap/services/monitor/pre-render.sh]: call kubectl directly doesn't work. please use ./deploy.py kubectl ***

**Comment by hongzhili** [comment]
How do I launch the new monitor system? still ./deploy.py kubernetes start monitor?

File: src/ClusterBootstrap/services/monitor/pre-render.sh (hunk 2 above)
Comment (xudifsd) [reply src/ClusterBootstrap/services/monitor/pre-render.sh]: This is used to generate configmap yaml file through files, it doesn't need to be called via `deploy.py kubectl`.

**Comment by xudifsd** [comment]
@hongzhili yes.

**Comment by hongzhili** [comment]
Should I start to merge this branch? does it still WIP?  @xudifsd

**Comment by xudifsd** [comment]
Possible problem: grafana still use dns to locate prometheus service. If dns has problems grafana will broken.

We can merge this and change it in future commits.

**Comment by xudifsd** [comment]
@hongzhili I have resolved the conflicts, and find a way to get static IP of prometheus.

You can merge this PR now.

File: src/ClusterBootstrap/deploy.py [hunk 3]
```diff
@@ -261,6 +261,8 @@ def update_config():
         config["elasticsearch_node"] = config["webportal_node"]
     if ("mysql_node" not in config):
         config["mysql_node"] = None if len(get_node_lists_for_service("mysql"))==0 else get_node_lists_for_service("mysql")[0]
+    if ("host" not in config["prometheus"]):
+        config["prometheus"]["host"] = None if len(get_node_lists_for_service("prometheus"))==0 else get_node_lists_for_service("prometheus")[0]
```
Comment (xudifsd) [inline src/ClusterBootstrap/deploy.py]: Just deployed in p40 bed, this config will become a public IP. How can I get a private IP?

**Pull request merged**